internal/docker/deployer.go

```diff
@@ -423,17 +423,15 @@ func waitForPorts(ctx context.Context, docker *client.Client, containerID string
 }
 
 // Waits until a homeserver deployment is ready to serve requests.
-func waitForContainer(ctx context.Context, docker *client.Client, hsDep *HomeserverDeployment, stopTime time.Time) (iterCount int, err error) {
-	var lastErr error = nil
-
+func waitForContainer(ctx context.Context, docker *client.Client, hsDep *HomeserverDeployment, stopTime time.Time) (iterCount int, lastErr error) {
```

Comment:
I am not great with Go so forgive my questions:
Does the function signature need to change now that it is no longer returning `iterCount` and `lastErr`? 
Also, it looks like `iterCount` is just incremented and doesn't do anything else, if it's not being returned anymore does it make sense to just ditch that variable?

Reply:
This function is using naked return values and named return args. See https://go.dev/tour/basics/7

So it is returning data, and by calling it `lastErr` you remove the need to ass `err = lastErr`.

Reply:
yup as kegsay says: we *are* still returning `iterCount` and `lastErr`, it's just happening automatically instead of explicitly.

https://go.dev/doc/effective_go#named-results explains this too.

(This is one of many things about Go I'm not entirely convinced about, but in some cases like this it does seem to be useful)

Reply:
Thanks all for the explanations, I appreciate it.